astropy/io/fits/tests/test_connect.py

```diff
@@ -304,6 +304,21 @@ def check_equal(filename, expected, start_from=1):
         check_equal(filename, 3, start_from=2)
         assert equal_data(t2, Table.read(filename, hdu=1))
 
+    def test_format_parse_on_read(self, tmpdir):
+        filename = str(tmpdir.join('test_format_parse_on_read.fits'))
+        primary_hdu = fits.PrimaryHDU()
+        c1 = fits.Column(name='a', array=np.array([b'1', b'2']), format='E')
+
+        table_hdu = fits.TableHDU.from_columns([primary_hdu, c1])
+        hdulist = fits.HDUList([table_hdu])
+        hdulist.writeto(filename)
+
+        with fits.open(filename) as hdulist:
+            data = hdulist[1].data
+
+            for col in data.columns:
+                assert np.isnan(col.array)
```

Comment:
This shows the failure, but would be more interesting to show a case that was failing before and works with the change, e.g. simply `Table.read(filename)` (I added a NaN to the array in the suggestion above).

Reply:
Ah, yes, of course. Changed, thanks!